### PR TITLE
Major version bump 5.0.0

### PR DESCRIPTION
**Pull request by cdhoffmann**
bumps the minimum deployment target to iOS 12 as per Xcode 15 requirements, and uses Xcode 15 in CI.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,7 +12,7 @@
 A simple, generic, extensible Rules Engine in Swift.
 
 ## Requirements
-- Xcode 14.1 (or newer)
+- Xcode 15 (or newer)
 - Swift 5.1 (or newer)
```
Comment (praveek) [inline README.md:16]: Can you update this to latest version?

**Review by praveek — APPROVED** [review]

**Comment by praveek** [comment]
Can you also remove the dependencies section from https://github.com/adobe/aepsdk-rulesengine-ios/blob/main/Package.swift#L11?

**Review by praveek — APPROVED** [review]

File: default.profraw
Comment (praveek) [inline default.profraw:1]: Was this commited by mistake?

Comment (cdhoffmann) [reply default.profraw:1]: No, must have happened when i was testing the llvm codecov stuff locally. Removed it.

**Review by praveek — APPROVED** [review]

**Pull request merged**